stdlib/http/client.pyi

```diff
@@ -99,6 +99,23 @@ responses: dict[int, str]
 
 class HTTPMessage(email.message.Message):
     def getallmatchingheaders(self, name: str) -> list[str]: ...  # undocumented
+    # override below all of Message's methods that use `_HeaderType` / `_HeaderTypeParam` with `str`
+    # avoiding having to make Message generic
+    def __getitem__(self, name: str) -> str: ...  # AnyOf[str, None]
```

Comment:
`str | None` should be correct here. A header is not guaranteed to exist, so not being prepared for a `None` value is a strong indication of a bug.

Reply:
That being said, the Any trick could be usable here (and in the referenced PR). I should do a primer run with it.

Reply:
I'm not sure I agree with that reasoning. While `get()` exists, `__getattr__` returns it as well. This is arguably a misfeature in the implementation (as it should just raise), but still that's something that a user should check against, because it absolutely can return `None`, even in unexpected scenarios.

Reply:
I'm still not happy about this. This should be an error in my opinion:

```python
def parse_headers(headers: SupportsGetItem[str]):
    header_value = headers["Content-Type"]
    full_header = f"Content-Type: {header_value}"

parse_headers(HTTPMessage(...))
```

Users of `HTTPMessage` must be equipped to get `None` values from `__getitem__`.